### PR TITLE
fix: include all semantic-release default branches for the release workflow + fix Genie

### PR DESCRIPTION
**Pull request by smoya**
**Description**

This PR adds all the default semantic-release branches to the release workflow for node apps.
See [semantic-release docs](https://semantic-release.gitbook.io/semantic-release/usage/configuration#branches).

This PR will also make Genie behave properly, who has been offering help randomly to people that didn't ask for it 😅 .

**Related issue(s)**
Relates to https://github.com/asyncapi/parser-js/pull/442#discussion_r783098429

### CODE REVIEW

File: .github/workflows/if-nodejs-release.yml [hunk 1]
```diff
@@ -10,7 +10,10 @@ on:
       # below lines are not enough to have release supported for these branches
       # make sure configuration of `semantic-release` package mentiones these branches
       - next
-      - '**-release'
+      - next-major
+      - beta
+      - alpha
+      - '**-release' # custom
```
Comment (smoya) [inline .github/workflows/if-nodejs-release.yml:16]: I didn't find this one on the semantic-release docs.

File: .github/workflows/if-nodejs-release.yml [hunk 2]
```diff
@@ -10,7 +10,10 @@ on:
       # below lines are not enough to have release supported for these branches
       # make sure configuration of `semantic-release` package mentiones these branches
       - next
-      - '**-release'
```
Comment (smoya) [inline .github/workflows/if-nodejs-release.yml:13]: The docs also show `'+([0-9])?(.{+([0-9]),x}).x` as regex for the branch name.I think we should also add it but waiting for feedback.

See https://semantic-release.gitbook.io/semantic-release/usage/configuration#branches

File: .github/workflows/if-nodejs-release.yml (hunk 1 above)
Comment (derberg) [reply .github/workflows/if-nodejs-release.yml:16]: this one is custom, driven by the discussion we had around release cycle and branch naming for specification where the branch name contains the release month.

File: .github/workflows/if-nodejs-release.yml (hunk 2 above)
Comment (derberg) [reply .github/workflows/if-nodejs-release.yml:13]: did you check if GH Actions support such regex, I don't think so afaik

Comment (smoya) [reply .github/workflows/if-nodejs-release.yml:13]: No it doesn't. It allows basic filtering but can't formulate something close to that regex. I would just skip this regex for now.
https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet

**Comment by smoya** [comment]
> lgtm,
> 
> please make additional bug fix not related to your change but just doesn't make sense to overload CI with tiny fixes, so we can add it here:
> 
> * https://github.com/asyncapi/.github/blob/master/.github/workflows/help-command.yml#L13 change from `github.event.issue.pull_request && contains(github.event.comment.body, '/help' && github.actor != 'asyncapi-bot') ` to `github.event.issue.pull_request && contains(github.event.comment.body, '/help') && github.actor != 'asyncapi-bot' `
> * https://github.com/asyncapi/.github/blob/master/.github/workflows/help-command.yml#L29 change from `${{ !github.event.issue.pull_request && contains(github.event.comment.body, '/help') && github.actor != 'asyncapi-bot'}}` to `!github.event.issue.pull_request && contains(github.event.comment.body, '/help') && github.actor != 'asyncapi-bot'`
> 
> 🙏🏼

Ok, time to sound picky here 😅 :
I don't think we should add unexpected changes to this PR. The reason is that the merged commit name will provide insights about what it is doing, and in case something goes wrong when merging, spotting the bug will become harder as some side effects where added. People won't know what really the changes were by reading the commit title and this is, IMHO, a bad practice.
I think we also want other people to don't mix changes on the same PR unless they are 100% related and described.

Therefore, I created a new PR for those changes you suggested here: https://github.com/asyncapi/.github/pull/124

**Comment by smoya** [comment]
> > lgtm,
> > please make additional bug fix not related to your change but just doesn't make sense to overload CI with tiny fixes, so we can add it here:
> > 
> > * https://github.com/asyncapi/.github/blob/master/.github/workflows/help-command.yml#L13 change from `github.event.issue.pull_request && contains(github.event.comment.body, '/help' && github.actor != 'asyncapi-bot') ` to `github.event.issue.pull_request && contains(github.event.comment.body, '/help') && github.actor != 'asyncapi-bot' `
> > * https://github.com/asyncapi/.github/blob/master/.github/workflows/help-command.yml#L29 change from `${{ !github.event.issue.pull_request && contains(github.event.comment.body, '/help') && github.actor != 'asyncapi-bot'}}` to `!github.event.issue.pull_request && contains(github.event.comment.body, '/help') && github.actor != 'asyncapi-bot'`
> > 
> > 🙏🏼
> 
> Ok, time to sound picky here 😅 : I don't think we should add unexpected changes to this PR. The reason is that the merged commit name will provide insights about what it is doing, and in case something goes wrong when merging, spotting the bug will become harder as some side effects where added. People won't know what really the changes were by reading the commit title and this is, IMHO, a bad practice. I think we also want other people to don't mix changes on the same PR unless they are 100% related and described.
> 
> Therefore, I created a new PR for those changes you suggested here: #124

As @derberg requested, I added the changes in this PR.

**Comment by derberg** [comment]
I also do not like mixing, just here is exception, not code but ci. We stop doing it once we get https://github.com/asyncapi/.github/issues/116 solved, so basically have solution to skip ci entirely in such PRs

**Review by derberg — COMMENTED** [review]
LGTM 🚀 

@KhudaDad414 anything from your side?

File: .github/workflows/help-command.yml [hunk 3]
```diff
@@ -26,7 +26,7 @@ jobs:
             - `/ready-to-merge` or `/rtm` - This comment will trigger automerge of PR in case all required checks are green, approvals in place and do-not-merge label is not added
             - `/do-not-merge` or `/dnm` - This comment will block automerging even if all conditions are met and ready-to-merge label is added
   create_help_comment_issue:
-    if: ${{ !github.event.issue.pull_request && contains(github.event.comment.body, '/help') && github.actor != 'asyncapi-bot'}}
```
Comment (KhudaDad414) [inline .github/workflows/help-command.yml:29]: @derberg are you sure about this one? It gives syntax error without the brackets, at least in my fork. https://github.com/KhudaDad414/.github/actions/runs/1717756142

**Review by KhudaDad414 — CHANGES_REQUESTED** [review]
LGTM.

File: .github/workflows/help-command.yml (hunk 3 above)
Comment (smoya) [reply .github/workflows/help-command.yml:29]: Yes, it does. Missing tag. I thought this was handled by github when running the workflow. Thoughts @derberg?

Comment (derberg) [reply .github/workflows/help-command.yml:29]: I'm pretty sure I read in docs it is fine, but have nothing against adding them, just to be save, but then add it to all ifs in this workflow please

Comment (smoya) [reply .github/workflows/help-command.yml]: I added back the syntax expression format `{{`` and `}}` on the `create_help_comment_issue`.

File: .github/workflows/if-nodejs-release.yml (hunk 2 above)
Comment (KhudaDad414) [reply .github/workflows/if-nodejs-release.yml:13]: @smoya thanks! can you change https://github.com/asyncapi/.github/blob/41d14e04e38d75a6123f898d894e36e42dfde72a/.github/workflows/help-command.yml#L13 to `${{ github.event.issue.pull_request && contains(github.event.comment.body, '/help' ) && github.actor != 'asyncapi-bot' }}` for consistency?

Comment (smoya) [reply .github/workflows/if-nodejs-release.yml:13]: done!

File: .github/workflows/help-command.yml [hunk 4]
```diff
@@ -10,7 +10,7 @@ on:
 
 jobs:
   create_help_comment_pr:
-    if: github.event.issue.pull_request && contains(github.event.comment.body, '/help' && github.actor != 'asyncapi-bot') 
+    if: ${{ github.event.issue.pull_request && contains(github.event.comment.body, '/help' && github.actor != 'asyncapi-bot') }}
```
Comment (KhudaDad414) [inline .github/workflows/help-command.yml:13]:
```suggestion
    if: ${{ github.event.issue.pull_request && contains(github.event.comment.body, '/help') && github.actor != 'asyncapi-bot' }}
```

**Review by KhudaDad414 — APPROVED** [review]

**Review by KhudaDad414 — APPROVED** [review]
LGTM!

**Comment by smoya** [comment]
Thanks folks for the review!

Anything else to add @derberg ? 🚀

**Review by derberg — APPROVED** [review]

**Pull request merged**